include/ginkgo/core/distributed/sparse_communicator.hpp

```diff
@@ -0,0 +1,89 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_PUBLIC_CORE_DISTRIBUTED_SPARSE_COMMUNICATOR_HPP_
+#define GKO_PUBLIC_CORE_DISTRIBUTED_SPARSE_COMMUNICATOR_HPP_
+
+
+#include <ginkgo/config.hpp>
+
+
+#if GINKGO_BUILD_MPI && GINKGO_HAVE_CXX17
+
+
+#include <variant>
+
+
+#include <ginkgo/core/base/dense_cache.hpp>
+#include <ginkgo/core/base/mpi.hpp>
+#include <ginkgo/core/distributed/index_map.hpp>
+
+
+namespace gko {
+namespace experimental {
+namespace distributed {
+
+
+/**
+ * Simplified MPI communicator that handles only neighborhood all-to-all
+ * communication.
+ */
+class sparse_communicator
```

Comment:
If it only provides a neighborhood all to all communication, then I think calling it a sparse communicator might be misleading.